source/util/convert_to_iso_string.ts

```diff
@@ -0,0 +1,20 @@
+function isIsoDate(str: string) {
+  if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(str)) return false;
+  const d = new Date(str);
+  return d instanceof Date && !isNaN(d.getTime()) && d.toISOString() === str; // valid date
+}
+
+export function convertToISOString(data: string | Date) {
```

Comment:
Can we add a docstring and tests for these methods?

Reply:
fixed!